fatiando/seismic/conv.py

```diff
@@ -0,0 +1,245 @@
+
+    * model : 2D-array
+        Vp values
+    * dt: float
+        Sample time of the ricker wavelet and of the resulting seismogram
+    * dz : float
+        Length of square grid cells
+    * rho : 2D-array (optional)
+        Density values for all the model, in depth domain.
+
+    Returns:
+
+    * vel_l : 2D-array
+        Velocity model in time domain
+    * rho_l : 2D-array
+        Density model in time domain
+
+    """
+    # downsampled time rate to make a better interpolation
+    n_samples, n_traces = [model.shape[0], model.shape[1]]
+    dt_dwn = dt/10.
+    if dt_dwn > dz/np.max(model):
+        dt_dwn = (dz/np.max(model))/10.
+    TWT = np.zeros((n_samples, n_traces))
+    TWT[0, :] = 2*dz/model[0, :]
+    for j in range(1, n_samples):
+        TWT[j, :] = TWT[j-1]+2*dz/model[j, :]
+    TMAX = max(TWT[-1, :])
+    TMIN = min(TWT[0, :])
+    TWT_rs = np.zeros(np.ceil(TMAX/dt_dwn))
+    for j in range(1, len(TWT_rs)):
+        TWT_rs[j] = TWT_rs[j-1]+dt_dwn
+    resmpl = int(dt/dt_dwn)
+    vel_l = _resampling(model, TMAX, TWT, TWT_rs, dt, dt_dwn, n_traces)
+    TWT_ts = np.zeros((np.ceil(TMAX/dt), n_traces))
+    for j in range(1, len(TWT_ts)):
+        TWT_ts[j, :] = TWT_rs[resmpl*j]
+    # density calculations
+    try:
+        rho_l = _resampling(rho, TMAX, TWT, TWT_rs, dt, dt_dwn, n_traces)
+    except TypeError:
+        rho_l = rho
+    return vel_l, rho_l
+
+
+def _resampling(model, TMAX, TWT, TWT_rs, dt, dt_dwn, n_traces):
+    """
+    Resamples the input data to adjust it after time conversion with the chosen
+    time sample rate, dt.
+
+    Returns:
+
+    * vel_l : resampled input data
+
+    """
+    vel = np.ones((np.ceil(TMAX/dt_dwn), n_traces))
+    for j in range(0, n_traces):
+        kk = np.ceil(TWT[0, j]/dt_dwn)
+        lim = np.ceil(TWT[-1, j]/dt_dwn)-1
+    # necessary do before resampling to have values in all points of time model
+        tck = interpolate.interp1d(TWT[:, j], model[:, j])
+        vel[kk:lim, j] = tck(TWT_rs[kk:lim])
+    # the model is extended in time because of depth time conversion
+        vel[lim:, j] = vel[lim-1, j]
+    # because of time conversion, the values between 0 e kk need to be filed
+        vel[0:kk, j] = model[0, j]
+    # resampling from dt_dwn to dt
+    vel_l = np.zeros((np.ceil(TMAX/dt), n_traces))
+    resmpl = int(dt/dt_dwn)
+    vel_l[0, :] = vel[0, :]
+    for j in range(0, n_traces):
+        for jj in range(1, int(np.ceil(TMAX/dt))):
+            vel_l[jj, j] = vel[resmpl*jj, j]  # 10=dt/dt_new, dt_new=0.002=2m
+    return vel_l
+
+
+def rickerwave(f, dt):
+    r"""
+    Given a frequency and time sampling rate, outputs ricker function. To
+    satisfy sampling and stability:
+
+    .. math::
+
+        f << \frac{1}{2 dt}.
+
+    Here, we consider this as:
+
+    .. math::
+
+        f < 0.2 \frac{1}{2 dt}.
+
+    Parameters:
+
+    * f : dominant frequency value in Hz
+    * dt : time sampling rate in seconds (usually it is in the order of ms)
```

Comment:
@victortxa please indicate the expected type of the input parameters (I'm assuming both are floats).